contracts/contracts/extensions/payout/SimplePayout.sol

```diff
@@ -0,0 +1,188 @@
+  /// @param _spentSalt the salt of the spent amount
+  /// @param _resultsCommitment the commitment of the results
+  /// @param _spentVoiceCreditsCommitment the commitment of the spent voice credits
+  /// @param _voteOptionTreeDepth the depth of the vote option tree
+  function payout(
+    uint256 _voteOptionIndex,
+    uint256 _spent,
+    uint256[][] calldata _proof,
+    uint256 _spentSalt,
+    uint256 _resultsCommitment,
+    uint256 _spentVoiceCreditsCommitment,
+    uint8 _voteOptionTreeDepth
+  ) external {
+    // check if the user has been paid already
+    if (paidUsers[_voteOptionIndex]) revert ProjectAlreadyPaid();
+    // set the vote option index as paid
+    paidUsers[_voteOptionIndex] = true;
+
+    // increment the number of paid users
+    unchecked {
+      paid++;
+    }
+
+    // get the address of the recipient
+    address recipient = recipientRegistry.getRecipient(_voteOptionIndex);
+
+    // we verify the proof
+    if (
+      !tally.verifyPerVOSpentVoiceCredits(
+        _voteOptionIndex,
+        _spent,
+        _proof,
+        _spentSalt,
+        _voteOptionTreeDepth,
+        _spentVoiceCreditsCommitment,
+        _resultsCommitment
+      )
+    ) revert InvalidProof();
+
+    // we need to calculate the amount that is to be given to the project (round down)
+    uint256 tokensToPay = (_spent * totalAmount) / totalVoiceCreditsSpent;
+
+    // transfer the token amount
+    // check whether is native token
+    if (payoutToken == address(0)) {
+      // transfer the native token amount
+      _handleNativeTransfer(recipient, tokensToPay);
+    } else {
+      // transfer the ERC20 token amount
+      IERC20(payoutToken).safeTransfer(recipient, tokensToPay);
+    }
+  }
+
+  /// @notice A function to collect the dust left from round downs
+  /// can only be called once all projects have been paid
+  function collectDust() external onlyOwner {
+    // check if all projects have been paid
+    if (recipientRegistry.getRecipientsNumber() != paid) revert NotAllProjectsPaid();
+
+    address token = payoutToken;
+    if (token == address(0)) {
+      // transfer the native token amount
+      payable(msg.sender).call{ value: address(this).balance }("");
+    } else {
+      // transfer the ERC20 token amount
+      IERC20(token).safeTransfer(msg.sender, IERC20(token).balanceOf(address(this)));
+    }
+  }
+
+  /// @notice Handle native transfers
+  /// @param _dest The destination address
+  /// @param _amount The amount to transfer
+  function _handleNativeTransfer(address _dest, uint256 _amount) internal {
+    // Handle Ether payment
+    if (address(this).balance < _amount) revert NotEnoughEther();
+    uint256 gas = gasleft();
+    (bool success, ) = _dest.call{ value: _amount, gas: gas }("");
+    // If the Ether transfer fails, wrap the Ether and try to send it as WETH.
+    if (!success) {
+      WETH.deposit{ value: _amount }();
+      IERC20(address(WETH)).safeTransfer(_dest, _amount);
+    }
+  }
```

Comment:
Is it possible if project from registry can be replaced with something different?
And is there a reason why you use `address#call`?

Reply:
not sure what exactly you are referring by "replaced with something different"?
I use call so there is more flexibility on what the receiver can do in their receive/fallback, and also on the sender side + I believe call is the recommended method of sending Ether nowawayds. I remember this particular pattern from my auditing days, as contracts can revert on receive and if they did and we don't handle that, we will never be able to complete the payout. 
Any other way you'd do this?

Reply:
I think it's fine regarding `call` so 
By "replaced with something different" I mean, if coordinator can add different project address or even user address to get the payout. So target project won't get anything because registry has different addresses.